crates/core/src/checkpoint.rs

```diff
@@ -208,6 +223,54 @@ where
         }))
     }
 
+    /// Calculate the median time past (MTP) for this checkpoint.
+    ///
+    /// Uses the 11 previous blocks (heights h-11 through h-1, where h is the current height)
+    /// to compute the MTP for the current block. This is used in Bitcoin's consensus rules
+    /// for time-based validations (BIP113).
+    ///
+    /// Note: This is a pseudo-median that doesn't average the two middle values.
+    ///
+    /// Returns `None` if the data type doesn't support block times or if any of the required
+    /// 11 sequential blocks are missing.
+    pub fn median_time_past(&self) -> Option<u32> {
+        let current_height = self.height();
+        let earliest_height = current_height.saturating_sub(Self::MTP_BLOCK_COUNT);
+        self._median_time_past(earliest_height..current_height)
+    }
```

Comment:
I'm curious if these new APIs could output wrong/invalid values if the blocks inserted have a wrong/invalid time field e.g malicious block source.

Reply:
Definitely. We are trusting the chain source here (and everywhere in BDK). Eventually, we should think about how to verify all consensus rules in BDK: difficulty, PoW, MTP, etc.